python-avd/pyavd/_eos_designs/structured_config/network_services/application_traffic_recognition.py

```diff
@@ -86,11 +75,16 @@ def _set_control_plane_application_profile(self: AvdStructuredConfigNetworkServi
                   - name: PFX-LOCAL-VTEP-IP
                     prefix_values: [Pathfinder vtep_ip]
         """
-        # Adding the application-profile
-        if self._wan_control_plane_application_profile_name in self.structured_config.application_traffic_recognition.application_profiles:
+        # Adding the application-profile if not overridden
+        if (
+            self.inputs.wan_virtual_topologies.control_plane_virtual_topology.application_profile
+            in self.structured_config.application_traffic_recognition.application_profiles
```

Comment:
I don't think we should inspect structured config like this. It makes things very fragile for future changes and much harder to track.

Reply:
thats how it is today. The goal here is to check if this has already been added or if we need to add the default one

Reply:
and yes it is fragile as moving things around made it break (used to be done in `application_traffic_recognition` contributor and now it has moved 💥 )

So maybe need a better way to track this but the expected behavior is:

"if this is not configured somewhere else then we need to add this"

Reply:
fixed!